src/main/java/bio/terra/service/filedata/FileMetadataUtils.java

```diff
@@ -186,4 +191,40 @@ public static Md5ValidationResult validateFileMd5ForIngest(
           cloudMd5, StringUtils.isEmpty(cloudMd5) ? Md5Type.NEITHER : Md5Type.CLOUD_PROVIDED);
     }
   }
+
+  public static List<FileModel> toFileModel(
+      List<FireStoreDirectoryEntry> directoryEntries,
+      List<FireStoreFile> files,
+      String collectionId) {
+    List<FileModel> resultList = new ArrayList<>();
+    if (directoryEntries.size() != files.size()) {
+      throw new FileSystemExecutionException("List sizes should be identical");
+    }
+
+    for (int i = 0; i < files.size(); i++) {
+      FireStoreFile file = files.get(i);
+      FireStoreDirectoryEntry entry = directoryEntries.get(i);
+
+      FileModel fileModel =
+          new FileModel()
+              .fileId(entry.getFileId())
+              .collectionId(collectionId)
+              .path(FileMetadataUtils.getFullPath(entry.getPath(), entry.getName()))
+              .size(file.getSize())
+              .created(file.getFileCreatedDate())
+              .description(file.getDescription())
+              .fileType(FileModelType.FILE)
+              .checksums(FileService.makeChecksums(file.getChecksumCrc32c(), file.getChecksumMd5()))
```

Comment:
OK, so I see why this can't call `FileService.makeChecksums(file)` (because `FireStoreFile` doesn't `extend FSItem`). If you wanted to fix that, you could make an interface that had the `getChecksum` methods that both these class implement, then you could make `makeChecksums()` accept that type instead of `FSItem`.